docs/registry/upload/_index.md

```diff
@@ -288,47 +290,120 @@ To update an existing module in the [Viam registry](https://app.viam.com/registr
    See the [GitHub actions documentation](https://docs.github.com/en/actions/creating-actions) for more information.
 
-1. Paste the following action template YAML into the edit window.
-
-   ```yaml {class="line-numbers linkable-line-numbers"}
-   on:
-     push:
-     release:
-       types: [released]
-
-   jobs:
-     publish:
-       runs-on: ubuntu-latest
-       steps:
-         - uses: actions/checkout@v3
-         - name: build
-           run: echo "your build command goes here" && false # <-- replace this with the command that builds your module's tar.gz
-         - uses: viamrobotics/upload-module@v1
-           # if: github.event_name == 'release' # <-- once the action is working, uncomment this so you only upload on release
-           with:
-             module-path: module.tar.gz
-             platform: linux/amd64 # <-- replace with your target architecture, or your module will not deploy
-             version: ${{ github.event_name == 'release' && github.ref_name || format('0.0.0-{0}.{1}', github.ref_name, github.run_number) }} # <-- see 'Versioning' section below for explanation
-             key-id: ${{ secrets.viam_key_id }}
-             key-value: ${{ secrets.viam_key_value }}
-   ```
+1. Select the `build-action` or `upload-module` action and paste the following action template YAML into the edit window.
+
+   {{< tabs >}}
+   {{% tab name="CI with build-action" %}}
+
+```yaml {class="line-numbers linkable-line-numbers"}
+# see https://github.com/viamrobotics/upload-module for help
+on:
+  push:
+    tags:
+      - "[0-9]+.[0-9]+.[0-9]+" # the build-action will trigger on tags with the format 1.0.0
+
+jobs:
+  publish:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: viamrobotics/build-action@v1
+        with:
+          # note: you can replace this line with 'version: ""' if
+          # you want to test the build process without deploying
+          version: ${{ github.ref_name }}
+          ref: ${{ github.sha }}
+          key-id: ${{ secrets.viam_key_id }}
+          key-value: ${{ secrets.viam_key_value }}
+```
+
+The `build-action` relies on a build command that you need to specify in the <file>meta.json</file> file that you created for your module when you first [uploaded it](/registry/upload/#upload-a-custom-module).
+At the end of your <file>meta.json</file>, add the build configuration:
+
+```json {class="line-numbers linkable-line-numbers" data-line="4-7"}
+{
+  "module_id": "example-module",
+  ...
+  "build": {
+    "build": "make module.tar.gz",
+    "arch" : ["linux/amd64", "linux/arm64", "darwin/arm64"]
+  }
+}
+```
+
+You can test this build configuration by running the following command on your development machine:
+
+```sh {class="command-line" data-prompt="$"}
+viam module build local
+```
+
+The command will run your build instructions without running a cloud build job.
```

Comment:
Supernit: `local` for what it does do, not just `cloud` for what it doesn't. Even though it's in the command name:

```suggestion
The command will run your build instructions locally without running a cloud build job.
```